Xamarin.PropertyEditing/ViewModels/RectanglePropertyViewModel.cs

```diff
@@ -1,14 +1,63 @@
 ﻿using System.Collections.Generic;
-using System.Drawing;
+using Xamarin.PropertyEditing.Drawing;
 
 namespace Xamarin.PropertyEditing.ViewModels
 {
 	internal class RectanglePropertyViewModel
-		: PropertyViewModel<Rectangle>
+		: PropertyViewModel<CommonRectangle>
 	{
 		public RectanglePropertyViewModel (IPropertyInfo property, IEnumerable<IObjectEditor> editors)
 			: base (property, editors)
 		{
 		}
+
+		public double X {
+			get { return Value.X; }
+			set {
+				if (Value.X == value)
+					return;
+
+				Value = new CommonRectangle (value, Value.Y, Value.Width, Value.Height);
+			}
+		}
+
+		public double Y {
+			get { return Value.Y; }
+			set {
+				if (Value.Y == value)
+					return;
+
+				Value = new CommonRectangle (Value.X, value, Value.Width, Value.Height);
+			}
+		}
+
+		public double Width {
+			get { return Value.Width; }
+			set {
+				if (Value.Width == value)
+					return;
+
+				Value = new CommonRectangle (Value.X, Value.Y, value, Value.Height);
+			}
+		}
+
+		public double Height {
+			get { return Value.Height; }
+			set {
+				if (Value.Height == value)
+					return;
+
+				Value = new CommonRectangle (Value.X, Value.Y, Value.Width, value);
+			}
+		}
+
+		protected override void OnValueChanged ()
```

Comment:
It looks like setting any of the four properties will trigger a change notification on all of them, through the setting of Value. On the other hand, we're not being more subtle on `Size`, so this is probably fine for now.

Reply:
@bleroy I didn't want to stray too much from the pattern already used in the CommonSize and CommonPoint editors where if one property was set, then both were set. Maybe go with this for now, we can always make it more granular later on, if we think it will give a better user experience?